api/src/main/java/org/openmrs/module/attachments/obs/Attachment.java

```diff
@@ -51,6 +51,8 @@ public Attachment(Obs obs) {
 		setDateTime(obs.getObsDatetime());
 		setComment(obs.getComment());
 		setComplexData(obs.getComplexData());
+		setBytesMimeType(getBytesMimeType());
+		setBytesContentFamily(getBytesContentFamily());
```

Comment:
That's not ok, the MIME type must be gotten from the obs's complex data. This is why I said that you will need a `ComplexDataHelper` at hand.

Once you have the MIME type it is trivial to infer the content family from it.

Reply:
hello @mks-d am able to fetch the MIME type with something like this `setBytesMimeType(complexDataHelper.getContentType(obs.getComplexData()));`
and when it comes to the contentFamily something like this `setBytesContentFamily(attachmentsContext.getContentFamily(complexDataHelper.getContentType(obs.getComplexData())));` which has forced me to come up with a new Attachment constructor ie `Attachment(Obs , ComplexDataHelper, AttachmentsContext )` coupling the context to the attachment......

Am requesting for your views on this approach that am taking  

Thanks.

Reply:
@gitcliff `AttachmentsContext#getContentFamily(String mimeType)` [is static](https://github.com/openmrs/openmrs-module-attachments/blob/9c61cf896be64bb09544287a567140df387b97dd/api/src/main/java/org/openmrs/module/attachments/AttachmentsContext.java#L346), so you can just use it inside of `Attachment`'s constructor. You don't actually need another constructor argument in that specific case.

Reply:
@mks-d thanks for your insights..

Reply:
@mks-d when i expand the existing constructor to what you recommended  ,,it [breaks](https://pastebin.com/FNpz1811) the existing unit tests int the `AttachmenServiceTest` class for those attachment instances that where taking obs only ..
A new attachment constructor with arguments obs and complexDataHelper solves the problem

Reply:
You mean [here](https://github.com/openmrs/openmrs-module-attachments/blob/9c61cf896be64bb09544287a567140df387b97dd/api-2.0/src/test/java/org/openmrs/module/attachments/AttachmentsServiceTest.java#L54)? Just autowire `ComplexDataHelper` in that class and use it there.

Reply:
yes @mks-d  i meant that,,,,
I have auto wired  `ComplexDataHelper` with `@Autowired `and `@Qualifier(AttachmentsConstants.COMPONENT_COMPLEXDATA_HELPER`) but still landing in the same `NullPointerException` when running the tests

Reply:
@gitcliff I need to see some code, just add commits to this PR.

Reply:
@mks-d i have pushed some code

Reply:
@mks-d am kindly requesting for your view on the code i pushed about the attachment constructor issue